File: java/src/main/java/benchmarks/BufferHashBenchmark.java
package benchmarks;

public abstract class BufferHashBenchmark extends Benchmark {
    protected byte[] data;
    private long resultVal;
    private long sizeVal;

    public BufferHashBenchmark() {
        resultVal = 0L;
        sizeVal = 0L;
    }

    @Override
    public void prepare() {
        if (sizeVal == 0) {
            sizeVal = configVal("size");
            data = new byte[(int) sizeVal];
            for (int i = 0; i < data.length; i++) {
                data[i] = (byte) Helper.nextInt(256);
            }
        }
    }

    abstract long test();

    @Override
    public void run(int iterationId) {
        resultVal += test();
    }

    @Override
    public long checksum() {
        return resultVal;
    }
}

File: kotlin/src/main/kotlin/benchmarks/BufferHashBenchmark.kt
package benchmarks

import Benchmark

abstract class BufferHashBenchmark : Benchmark() {
    protected val sizeVal = configInt("size")
    protected lateinit var data: ByteArray
    protected var resultVal: UInt = 0u

    override fun prepare() {
        data = ByteArray(sizeVal) { Helper.nextInt(256).toByte() }
    }

    abstract fun test(): UInt

    override fun run(iterationId: Int) {
        resultVal += test()
    }

    override fun checksum(): UInt = resultVal
}
